files/sdw-admin.py

```diff
@@ -270,6 +288,185 @@ def perform_uninstall():
+        [
+            "qvm-run",
+            "--pass-io",
+            "vault",
+            f"gpg --homedir {TAILS_GNUPG_PATH} -K --fingerprint --with-colon",
+        ],
+        text=True,
+    )
+    fingerprint = extract_fingerprint(gpg_output)
+    if not fingerprint:
+        raise SDWAdminException("Error reading submission key fingerprint")
+
+    gpg_privkey = subprocess.check_output(
+        [
+            "qvm-run",
+            "--pass-io",
+            "vault",
+            f"gpg --homedir {TAILS_GNUPG_PATH} --export-secret-keys --armor {fingerprint}",
+        ],
+        text=True,
+    )
+
+    temp_file = "/tmp/sd-journalist.sec"
+    with open(temp_file, "w") as f:
+        f.write(gpg_privkey)
+
+    subprocess.check_call(["sudo", "cp", temp_file, SCRIPTS_PATH])
+
+    return fingerprint
+
+
+def import_journalist_interface_config():
+    """
+    Imports Journalist Interface address and authentication info from USB drive to dom0.
+    Assumes that USB drive is attached to vault VM and decrypted.
+    Returns (hostname, key) of the journalist interface hidserv
+    """
+    journalist_interface_config = subprocess.check_output(
+        [
+            "qvm-run",
+            "--pass-io",
+            "vault",
+            f"cat {TAILS_JOURNALIST_INTERFACE_CONFIG}",
+        ],
+        text=True,
+    )
+    fields = journalist_interface_config.strip().split(":")
+    addr = fields[0]
+    auth_token = fields[3]
+    return addr, auth_token
+
+
+def import_config():
+    submission_key_fingerprint = _try_read_submission_key()
+    if not submission_key_fingerprint:
+        subprocess.Popen(
+            ["qvm-start", "vault"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+        )
+        print(
+            "Importing SecureDrop submission key from USB...\n\n\n"
+            "Ensure that USB containing submission key is connected.\n\n"
+            "1. Attach the USB to the vault VM\n"
+            "2. Open Thunar File Manager in the vault VM\n"
+            "3. Select the USB drive in the left sidebar of the file manager.\n"
+            "It should be listed under Devices as 'N GB Encrypted'.\n"
+            "Enter the correct passphrase when prompted.\n\n"
+        )
+        response = input("Are you ready to proceed (y/N)?")
+        if response.lower() != "y":
+            print("Exiting.")
+            return
+        print("Importing submission key...")
+        submission_key_fingerprint = import_submission_key()
+        print(
+            "Submission key import complete!\n"
+            "Please detach and disconnect the USB containing the submission key\n\n"
+        )
+    else:
+        print("Found submission key file, proceeding")
+
+    try:
+        validate_config(SCRIPTS_PATH)
+        print("Valid configuration found, configuration complete")
+    except SDWAdminException:
+        subprocess.Popen(
+            ["qvm-start", "vault"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
```

Comment:
One other option here could be to spin up a non-networked dispvm instead, but it might be overkill.